src/main/kotlin/com/fasterxml/jackson/module/kotlin/Exceptions.kt

```diff
@@ -1,10 +1,14 @@
 package com.fasterxml.jackson.module.kotlin
 
+import com.fasterxml.jackson.core.JsonParser
 import com.fasterxml.jackson.databind.JsonMappingException
-import java.io.Closeable
+import com.fasterxml.jackson.databind.exc.MismatchedInputException
 import kotlin.reflect.KParameter
 
 /**
- * Specialized [JsonMappingException] sub-class used to indicate that a mandatory Kotlin constructor parameter was missing or null.
+ * Specialized [JsonMappingException] sub-class used to indicate that a mandatory Kotlin constructor
+ * parameter was missing or null.
  */
-class MissingKotlinParameterException(val parameter: KParameter, val processor: Closeable? = null, val msg: String) : JsonMappingException(processor, msg)
\ No newline at end of file
+class MissingKotlinParameterException(val parameter: KParameter,
```

Comment:
Only one minor concern wrt backwards compatibility: I think this would change bytecode constructor signature, so theoretically could be binary incompatible (even if mostly source compatible?).
Not sure if worth worrying about but thought I'd mention it (I realize that change itself needed due to base class constructor difference). Also not sure if there's an easy way to add alternate constructor overload, in which `Closeable` is just forcibly cast to `JsonParser` (which must have been already passed)

Reply:
Very good point, I think adding that secondary constructor is all that is needed.

Reply:
Sounds good! Also: mark the legacy one `@Deprecated` for 2.x to keep it there; then can remove from 3.0 (`master`) since 3.x will not be backwards binary-compatible anyway and all deprecated methods can (and most should) be removed.

Reply:
👍 Thanks for the tips, you've clearly got experience writing libraries ;-)

I rebased the branche, force pushed, and pointed it at 2.11.

Reply:
Cool! And one last thing: could you update `release-notes/VERSION-2.x` and, if relevant, `release-notes/CREDITS-2.x` (esp. if issue/suggestion was by someone outside dev team I'll try to add a note as to who contributed what, least we can do :) )

Now that I remember, I will actually add you as a co-author in CREDITS as well.

Reply:
Great, thanks.  Version & credits updated, should be all set.